### PR TITLE
Refactor for distutils, python-3.x, and WARNING support

### PR DESCRIPTION
**Pull request by orlitzky**
I started adding a setup.py file for distutils and got a little carried away. This does a few big things. I've kept the commits atomic, but it might be easier to just review from scratch:

  * A new setup.py file for distribution/installation
  * Switched to python-3.5+ (subprocess API changed dictated that lower bound).
  * Added a test suite.
  * Refactored the script into an executable and library. This was necessary to avoid the ".py" extension on the installed executable while also keeping the test suite working.
  * Added support for WARNING results, and (as a result) changed the mode names from one/all to best/worst.

I'm sure I'll find more stuff to fix, but for now I think we can make a 0.0.1 release and I'll get it packaged in Gentoo so that we can deploy it "for real."

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,64 @@
...
-  commands          Checks to run. Enclose (separately) in quotes.
+  command              check to run, enclosed in quotes
 
 optional arguments:
-  -h, --help        show this help message and exit
-  --mode {all,one}  Specify `one` or `all`. `one` will return OK if any one of
-                    the checks is successfull. `all` will return CRITICAL if
-                    any one of the checks fail. default is `all`.
+  -h, --help           show this help message and exit
+  --mode {worst,best}  which individual check result should be the overall
```
Comment (clarkbox) [inline README.md:40]: i never was satisfied with "all/one". it is not clear what it means, and there is no precedent for the pattern. is there a reason you have used "best/worst"? maybe "all/any", which would be analogous with python's all() and any()?

another thought is to change the param to a bool, "optimistic". if false (default), return success if all pass. if true: return success if any pass). the downside to this solution is that we would have to change back to a string if we every want to add more than two "modes", whereas above we could add "none", "one" down the road if needed...

in any case this is fine, just doing some bikeshedding.

Comment (orlitzky) [reply README.md:40]: "Any" and "all" no longer match up nicely to the possible outputs. I added the ability to report warnings from the individual checks, and to return WARNING as a result. The problem I then ran into was that the names "all" or "one" don't let you specify what to do when you see WARNING.

With "best," for example, you can say for example that if you see CRITICAL/WARNING, then you want to report WARNING. Likewise with "worst" you can return WARNING from WARNING/OK. This is "backwards compatible" in the sense that it still matches up with the previous behavior if all you ever see is CRITICAL and/or OK, but the names are now more descriptive when WARNING is involved. For the same reason, "any" and "all" are confusing in that context. Quick, what should "any" return when you see CRITICAL/WARNING? I have to stop and think; but with best/worst it's immediately clear.

As far as I can see, an "optimistic" flag would work though. I just worry that the documentation for that option is going to involve the words "best" and "worst" =)

Comment (clarkbox) [reply README.md:40]: ok good enough for me.

**Comment by orlitzky** [comment]
Thank you! Can I bother you one last time to tag a v0.0.1 and upload the result of `python setup.py sdist` to the corresponding release? That should be the final piece of the puzzle.

**Comment by orlitzky** [comment]
> Thank you! Can I bother you one last time to tag a v0.0.1 and upload the result of `python setup.py sdist` to the corresponding release? That should be the final piece of the puzzle.

ping =)

I hate to be annoying, but I've got everything ready to go now. However if I add this to gentoo as-is, [I have to give it an ugly name with "_pre" in it.](https://devmanual.gentoo.org/ebuild-writing/file-format/index.html) to indicate that it may not be the final 0.0.1 release.

**Comment by clarkbox** [comment]
Hey sorry this slipped my mind. Will get it done tomorrow.

**Comment by clarkbox** [comment]
Hey, getting attacked by the day job... haven't forgotten...

**Comment by clarkbox** [comment]
Ok release up! Let me know how it looks/if I missed anything.

https://github.com/clarkbox/check_multiple/releases

**Comment by orlitzky** [comment]
Everything looks good here. Committed to Gentoo a few minutes ago: https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=37d33e80d0a18481

Thanks for all your help resurrecting this.

**Comment by clarkbox** [comment]
Great. Thanks for all the work you did getting it into Gentoo. Let me know
if I can help in the future.

**Pull request merged**